exporter/tanzuobservabilityexporter/metrics.go

```diff
@@ -249,16 +282,64 @@ func (g *gaugeConsumer) PushInternalMetrics(errs *[]error) {
 	}
 }
 
-func (g *gaugeConsumer) pushSingleNumberDataPoint(
-	metric pdata.Metric, numberDataPoint pdata.NumberDataPoint, errs *[]error) {
+type sumConsumer struct {
+	sender                senders.MetricSender
+	logger                *zap.Logger
+	missingValues         counter
+	reportInternalMetrics bool
+}
+
+func newSumConsumer(sender senders.MetricSender, options *consumerOptions) typedMetricConsumer {
+	var fixedOptions consumerOptions
+	if options != nil {
+		fixedOptions = *options
+	}
+	fixedOptions.replaceZeroFieldsWithDefaults()
+	return &sumConsumer{
+		sender:                sender,
+		logger:                fixedOptions.Logger,
+		reportInternalMetrics: fixedOptions.ReportInternalMetrics,
+	}
+}
+
+func (s *sumConsumer) Type() pdata.MetricDataType {
+	return pdata.MetricDataTypeSum
+}
+
+func (s *sumConsumer) Consume(metric pdata.Metric, errs *[]error) {
+	sum := metric.Sum()
+	isDelta := sum.AggregationTemporality() == pdata.MetricAggregationTemporalityDelta
+	numberDataPoints := sum.DataPoints()
+	for i := 0; i < numberDataPoints.Len(); i++ {
+		// If sum metric is a delta type, send it to tanzu observability as a
+		// delta counter. Otherwise, send it to tanzu observability as a gauge
+		// metric.
+		if isDelta {
+			s.pushSingleNumberDataPoint(metric, numberDataPoints.At(i), errs)
+		} else {
+			pushGaugeSingleNumberDataPoint(
+				metric, numberDataPoints.At(i), errs, s.sender, s.logger, &s.missingValues)
+		}
+	}
+}
+
+func (s *sumConsumer) PushInternalMetrics(errs *[]error) {
+	if s.reportInternalMetrics {
+		s.missingValues.Report(missingValueMetricName, typeIsSumTags, s.sender, errs)
+	}
+}
+
+func (s *sumConsumer) pushSingleNumberDataPoint(
```

Comment:
For consistency with the Gauge pusher, rename to `pushNumberDataPoint`.

Reply:
Done.